src/common/dom/deep-active-element.ts

```diff
@@ -1,8 +1,47 @@
-export const deepActiveElement = (
-  root: DocumentOrShadowRoot = document
-): Element | null => {
-  if (root.activeElement?.shadowRoot?.activeElement) {
-    return deepActiveElement(root.activeElement.shadowRoot);
-  }
-  return root.activeElement;
+import { HaDialog } from "../../components/ha-dialog";
+import { HaButtonMenu } from "../../components/ha-button-menu";
+
+export const deepActiveElement = <
```

Comment:
I'm not following your suggestion.  Do you mean have the dialog manager call a new function that calls the original `deepActiveElement`?  It's impossible to use the original version because the checks need to be done at the document and every shadow root level.